crates/cli/src/opts/build/utils.rs

```diff
@@ -45,66 +42,53 @@ pub fn solar_pcx_from_build_opts<'sess>(
         None => project.paths.read_input_files()?,
     };
 
-    // Only process sources with latest Solidity version to avoid conflicts.
-    let graph = Graph::<MultiCompilerParsedSource>::resolve_sources(&project.paths, sources)?;
```

Comment:
see https://github.com/foundry-rs/foundry/pull/11371/commits/58c5561761527352519b3eabe4e1fa79d91509e6

Reply:
reverted in 18947ee259, this is related to multi-version compilations which is probably untested so I'll look at it at a later time